mobile/src/main/java/org/openhab/habdroid/core/connection/AbstractConnection.java

```diff
@@ -97,24 +98,60 @@ public String getOpenHABUrl() {
     }
 
     @Override
-    public boolean checkReachabilityInBackground() {
+    public boolean checkReachability() {
         Log.d(TAG, "Checking reachability of " + getOpenHABUrl());
+
+        Socket s;
         try {
-            URL url = new URL(getOpenHABUrl());
-            int checkPort = url.getPort();
-            if (url.getProtocol().equals("http") && checkPort == -1)
-                checkPort = 80;
-            if (url.getProtocol().equals("https") && checkPort == -1)
-                checkPort = 443;
-            Socket s = new Socket();
-            s.connect(new InetSocketAddress(url.getHost(), checkPort), 1000);
-            Log.d(TAG, "Socket connected");
-            s.close();
-            return true;
-        } catch (Exception e) {
-            Log.e(TAG, e.getMessage());
+            s = createConnectedSocket();
+        } catch (MalformedURLException e) {
+            Log.d(TAG, "URL could not be parsed", e);
             return false;
         }
+
+        if (s == null) {
+            return false;
+        }
+
+        Log.d(TAG, "Socket connected");
+        try {
+            s.close();
+        } catch (IOException e) {
+            Log.d(TAG, "Could not close socket connection.", e);
```

Comment:
That log IMHO is useless. There's nothing we can do anyway, and it's not like we or the user would care.